### ISSUE TITLE
Add option to suppress `PLR6301` for custom instance method decorators

### ISSUE THREAD

**Issue by mrmeszaros** [astral-sh/ruff#27448]
### Context

- `PLR6301` already respects `@typing.override` because the override contract (not the method body) determines whether self is meaningful
- the same reasoning could apply to any decorator that changes the calling convention
- `self` may be required by contract even if the body never references it

### Problem

- there is no way to extend this exemption to user-defined decorators
- sometimes a method's signature is part of its contract with the decorator, not just an implementation detail
- converting such a method to `@staticmethod` would silently break that contract, but `PLR6301` has no way to know this
- `PLR6301` fires a false positive on such methods

### Example

```python
import inspect

def handler(event_type):
    def decorator(fn):
        fn.event_type = event_type
        return fn
    return decorator

class BaseHandler:
    def __init_subclass__(cls, **kwargs):
        super().__init_subclass__(**kwargs)
        cls.registry = dict(getattr(cls, 'registry', {}))
        for _, fn in inspect.getmembers(cls, predicate=inspect.isfunction):
            if hasattr(fn, 'event_type'):
                cls.registry[fn.event_type] = fn
    
    def receive(self, event) -> None:
        fn = self.registry.get(type(event))
        if fn is not None:
            fn(self, event)

class CustomEvent:
    pass

class SomeHandler(BaseHandler):
    @handler(CustomEvent)
    def on_custom(self, event: CustomEvent) -> None:
        print(event)
```

- ruff check fires `PLR6301` on `on_custom` even though self cannot be removed without breaking the dispatch protocol

### Requested behaviour

A new `[tool.ruff.lint.pylint]` option, analogous to the existing `pep8-naming.classmethod-decorators` and `pep8-naming.staticmethod-decorators`, that suppresses `PLR6301` for methods carrying specified decorators:

```toml
[tool.ruff.lint.pylint]
instance-method-decorators = ["handler"]  # suppress PLR6301 for these decorators
```

### Workarounds today

- `# noqa: PLR6301` on every affected method  - noisy
- turning off `PLR6301` -  hides genuine violations

### Related

- #19396 (same decorator-awareness gap)

### Version

ruff 0.15.12
